crates/polars-io/src/file_cache/utils.rs

```diff
@@ -95,7 +95,7 @@ pub fn init_entries_from_uri_list(
                         let cloud_path = object_path_from_str(&prefix)?;
 
                         let object_store =
-                            object_stores[std::cmp::min(i, object_stores.len())].clone();
+                            object_stores[std::cmp::min(i, object_stores.len() - 1)].clone();
```

Comment:
silly mistake 😅

we unfortunately can't add this to our test suite because it needs a real bucket

Reply:
Happens to the best of us 😅

Reply:
Thanks for the quick fix!

Reply:
@nameexhaustion  Would it make sense to add a test which uses Azurite (using docker) or similar to test reading from "the cloud"?  https://github.com/Azure/Azurite